script/watch.js

```diff
@@ -0,0 +1,170 @@
+// babel-register only looks for environment variables
+process.env.BABEL_ENV = process.env.BABEL_ENV || 'test';
+// use babel-register to compile files on the fly
+require('babel-register');
+
+// require mocha setup files
+require('../test/util/mocha-setup.js');
+require('../test/util/enzyme-setup.js');
+
+const { Writable } = require('stream');
+const chokidar = require('chokidar');
+const CLIEngine = require('eslint').CLIEngine;
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const glob = util.promisify(require('glob'));
+
+const eslint = new CLIEngine();
+const formatter = eslint.getFormatter('stylish');
+const sasslinter = require('sass-lint');
+const Mocha = require('mocha');
+
+const chalk = require('chalk');
+const path = require('path');
+const _ = require('lodash');
+
+// no arguments, run unit tests
+// any arguments provited then tests run must be explicitly provided
+let options;
+if (process.argv.length === 2) {
+  options= {
+    eslint: false,
+    sasslint: false,
+    showErrors: false,
+    unit: true
+  };
+} else {
+  options = {
+    eslint: process.argv.includes('eslint'),
+    sasslint: process.argv.includes('sasslint'),
+    showErrors: process.argv.includes('showErrors'),
+    unit: process.argv.includes('unit')
+  };
+}
+
+const watchArray = [
+  './src/**/*.jsx',
+  './src/**/*.js',
+  './test/**/*.unit.spec.*'
+];
+
+if (options.sassLint) {
+  watchArray.push('./src/**/*.scss');
+}
+
+const srcWatcher = chokidar.watch(watchArray);
```

Comment:
I wonder if something like this is doable? https://github.com/mochajs/mocha/issues/2763

Reply:
Though I suppose that wouldn't tell us which tests to re-run.

Reply:
This is refining the `delete` approach to only uncache the files that have changed. This is cleaner and I like it more- right now there's no direct link between what mocha requires and what gets cached- I'm relying on using similar glob patterns and executing these two tasks independently. This would explicitly track what mocha is requiring and only decache those files.

There's something in this example about doing it recursively that I don't understand. I was just deleting the file path key from the require cache. 👀

Reply:
What is this syntax? it's not CoffeeScript. 🙄